metrics/config.go

```diff
@@ -235,14 +241,24 @@ func createMetricsConfig(ops ExporterOptions, logger *zap.SugaredLogger) (*metri
 		if !allowCustomMetrics {
 			servingOrEventing := metricskey.KnativeRevisionMetrics.Union(
 				metricskey.KnativeTriggerMetrics).Union(metricskey.KnativeBrokerMetrics)
-			mc.recorder = func(ctx context.Context, ms stats.Measurement, ros ...stats.Options) error {
-				metricType := path.Join(mc.stackdriverMetricTypePrefix, ms.Measure().Name())
-
-				if servingOrEventing.Has(metricType) {
-					return stats.RecordWithOptions(ctx, append(ros, stats.WithMeasurements(ms))...)
+			mc.recorder = func(ctx context.Context, mss []stats.Measurement, ros ...stats.Options) error {
+				// Perform array filtering in place using two indices: w(rite)Index and r(ead)Index.
+				wIdx := 0
+				for rIdx := 0; rIdx < len(mss); rIdx++ {
+					metricType := path.Join(mc.stackdriverMetricTypePrefix, mss[rIdx].Measure().Name())
+					if servingOrEventing.Has(metricType) {
+						mss[wIdx] = mss[rIdx]
+						wIdx++
+					}
+					// Otherwise, skip the measurement (because it won't be accepted).
+				}
+				// Trim the list to the number of written objects.
+				mss = mss[:wIdx]
```

Comment:
Do you need to do this reassignment here, or just do the slice on line 261?

Reply:
Mostly for readability, I don't need reassignment

Reply:
`mess[:wIdx]...` is a bit hard to comprehend from the getgo.